### PR TITLE
Robutts slur and stammer robotically

### PR DESCRIPTION
**Pull request by FlattestGuitar**
:cl: FlattestGuitar
tweak: Robots now slur and stutter robotically
/:cl:

I'm not 100% sure if people want this to happen or not, so feel free to discuss.

Stammering IPC

Slurring IPC

### CODE REVIEW

**Comment by Kevak** [comment]
All the yes.

**Comment by NTSAM** [comment]
That stuttering is even more unreadable than normal stuttering
:+1: anyway

**Comment by KasparoVy** [comment]
Easy to read if you just glance through it haha, looks good. Nice idea

**Comment by FlattestGuitar** [comment]
The hilarious part is that it doesn't cut away any information normal stuttering wouldn't. It just adds the weird characters where there normally would be a repeated letter, but somehow this is indeed more unreadable.

**Comment by Spacemanspark** [comment]
:+1: Hilarious.

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -346,6 +346,29 @@ proc/slur(phrase)
...
 	return newphrase
 
+proc/roboslur(phrase) //for robutts
+	phrase = html_decode(phrase)
+	var/leng=lentext(phrase)
+	var/counter=lentext(phrase)
+	var/newphrase=""
+	var/newletter=""
+	while(counter>=1)
+		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
```
Comment (tigercat2000) [inline code/modules/mob/mob_helpers.dm]: 0+1, 0+2? what

**Comment by pinatacolada** [comment]
:+1: :+1: :+1:

**Comment by sasanek12** [comment]
:+1:

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/mob/mob_helpers.dm]: ```counter``` gets smaller by 1 every iteration.
so while it starts at 0+1, 0+2
it slowly becomes:
1+1, 1+2
2+1, 2+2
3+1, 3+2
as ```leng``` is never changed.

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -346,6 +346,29 @@ proc/slur(phrase)
...
+			if(lowertext(newletter)=="a")	newletter="ah"
+			if(lowertext(newletter)=="c")	newletter="k"
+		switch(rand(1,15))
+			if(1,3,5,)	newletter="[lowertext(newletter)]"
+			if(2,4,6,)	newletter="[uppertext(newletter)]"
+			if(7, 8, 15)	newletter+=pick("@", "!", "#", "$", "%", "&", "?")
+			//if(9,10)	newletter="<b>[newletter]</b>"
+			//if(11,12)	newletter="<big>[newletter]</big>"
+			//if(13)	newletter="<small>[newletter]</small>"
+		newphrase+="[newletter]";counter-=1
```
Comment (tigercat2000) [inline code/modules/mob/mob_helpers.dm]: oh, i see how i missed `counter -= 1`
please no semicolons, move this to a new line

Comment (RemieRichards) [inline code/modules/mob/mob_helpers.dm]: sadly, the original does this too.

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (FlattestGuitar) [inline code/modules/mob/mob_helpers.dm]: Good point, I'll do that.

**Comment by FlattestGuitar** [comment]
I'm not sure if there's any proper way to make the robostutter be replaced by a variable under the normal stutter, since it does different things to the text rather than just change a variable.

**Comment by taukausanake** [comment]
And here I thought at some point we had buttbots ported over. I am disappoint

**Comment by Fox-McCloud** [comment]
>And here I thought at some point we had buttbots ported over. I am disappoint

Going to need a forum post+discussion about that one.

**Comment by Spacemanspark** [comment]
no

**Comment by KasparoVy** [comment]
Some people get a real buttache over the buttbots that I'll never underbutt

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -70,7 +70,7 @@ proc/get_radio_key_from_channel(var/channel)
 /mob/living/proc/handle_speech_problems(var/message, var/verb)
 	var/list/returns[3]
 	var/speech_problem_flag = 0
-
+	var/robot = isSynthetic()
```
Comment (TheDZD) [inline code/modules/mob/living/say.dm]: Huh, I wasn't aware that BYOND actually allowed you to do this.

Comment (Fox-McCloud) [inline code/modules/mob/living/say.dm]: some things, you can.

For example, reagent datums have a color, which you can defined as a hex string text OR `rgb(0,0,0)`

Comment (PJB3005) [inline code/modules/mob/living/say.dm]: Yeah, build in BYOND procs which always have the same output and have constant args can be used like that.

**Pull request merged**